### ISSUE TITLE
[feat] support for DeepseekV2

### ISSUE THREAD

**Issue by tmm1** [linkedin/Liger-Kernel#129]
### 🚀 The feature, motivation and pitch

It would be nice to support DeepseekV2 models. Unfortunately the modeling code is not yet accepted into transformers, and requires trust_remote_code=True

I'm monkey-patching myself for now, and wanted to leave some notes that may be helpful when support is added officially down the road.

One initial issue when swapping in swiglu:

```
  File "/mnt/ML/huggingface/modules/transformers_modules/deepseek-ai/DeepSeek-Coder-V2-Lite-Base/ea9b066cee82f82906fdd58898cb3788b1c5d770/modeling_deepseek.py", line 555, in <listcomp>
    DeepseekV2MLP(
TypeError: LigerSwiGLUMLP.__init__() got an unexpected keyword argument 'intermediate_size'
```

**Comment by tmm1** [comment]
> ```
> modeling_mod.apply_rotary_pos_emb = liger_rotary_pos_emb
> ```

this is causing loss calculations to be wildly different for some reason

i will investigate further

---

> TypeError: LigerSwiGLUMLP.__init__() got an unexpected keyword argument 'intermediate_size'

i was able to fix this issue as follows:

```python
modeling_mod.DeepseekV2MLP.forward = LigerSwiGLUMLP.forward
```

**Comment by xinyubai1209** [comment]
deepseek v2 use MLA(Multi-head Latent Attention) to reduce the kv cache.

**Comment by qingquansong** [comment]
Yeah, deepseekv2 one is quite interesting as it used decoupled RoPE. 

For the MLA part, since it mainly target on inference case speed up with absorbed low-rank projection matrices into the original linear matrices. Feel free to first try implementing the layers apart from that first and can gradually improve with separate prs. Thanks for the interesting feature request and rapid kick off~
